From 6079761aa63b9be9de73e89bb23c8440c87cf5b3 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Fri, 30 Jan 2015 12:01:44 +0800
Subject: [PATCH] driver: pm: move invoking act8865_set_power_saving_mode() to
 the file pm.c

Move invoking act8865_set_power_saving_mode(void) from the board file
to the file pm.c to decrease the duplicated code.

Add the config to the selet PMIC power-saving mode.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 Config.in                                 | 9 +++++++++
 board/sama5d3_xplained/sama5d3_xplained.c | 3 ---
 board/sama5d4_xplained/sama5d4_xplained.c | 3 ---
 board/sama5d4ek/sama5d4ek.c               | 3 ---
 driver/act8865.c                          | 2 ++
 driver/driver_cpp.mk                      | 4 ++++
 driver/pm.c                               | 4 ++++
 7 files changed, 19 insertions(+), 9 deletions(-)

diff --git a/Config.in b/Config.in
index fd80dab..790135d 100644
--- a/Config.in
+++ b/Config.in
@@ -473,6 +473,15 @@ config	CONFIG_CODEC_ON_TWI3
 
 endchoice
 
+config CONFIG_PM_PMIC
+	bool "PMIC (ACT8865)"
+	depends on CONFIG_HAS_PMIC_ACT8865
+	select CONFIG_ACT8865
+	default y
+	help
+	  This interface allow you to select the PMIC(ACT8865) chip on the board
+	  which will be set the Power-Saving mode.
+
 endmenu
 
 menu "Board's Workaround Options"
diff --git a/board/sama5d3_xplained/sama5d3_xplained.c b/board/sama5d3_xplained/sama5d3_xplained.c
index e8d790a..778cdf8 100644
--- a/board/sama5d3_xplained/sama5d3_xplained.c
+++ b/board/sama5d3_xplained/sama5d3_xplained.c
@@ -419,9 +419,6 @@ void hw_init(void)
 #endif
 
 #ifdef CONFIG_ACT8865
-	/* Set ACT8865 REG power saving mode */
-	act8865_set_power_saving_mode();
-
 	/* Set ACT8865 output voltage */
 	sama5d3ek_act8865_set_reg_voltage();
 
diff --git a/board/sama5d4_xplained/sama5d4_xplained.c b/board/sama5d4_xplained/sama5d4_xplained.c
index d8b9ece..6405fa3 100644
--- a/board/sama5d4_xplained/sama5d4_xplained.c
+++ b/board/sama5d4_xplained/sama5d4_xplained.c
@@ -664,9 +664,6 @@ void hw_init(void)
 #endif
 
 #ifdef CONFIG_ACT8865
-	/* Set ACT8865 REG power saving mode */
-	act8865_set_power_saving_mode();
-
 	/* Set ACT8865 output voltage */
 	sama5d4ek_act8865_set_reg_voltage();
 
diff --git a/board/sama5d4ek/sama5d4ek.c b/board/sama5d4ek/sama5d4ek.c
index 868a1c9..af32346 100644
--- a/board/sama5d4ek/sama5d4ek.c
+++ b/board/sama5d4ek/sama5d4ek.c
@@ -702,9 +702,6 @@ void hw_init(void)
 #endif
 
 #ifdef CONFIG_ACT8865
-	/* Set ACT8865 REG power saving mode */
-	act8865_set_power_saving_mode();
-
 	/* Set ACT8865 output voltage */
 	sama5d4ek_act8865_set_reg_voltage();
 
diff --git a/driver/act8865.c b/driver/act8865.c
index 6928905..d567578 100644
--- a/driver/act8865.c
+++ b/driver/act8865.c
@@ -132,6 +132,7 @@ int act8865_set_reg_voltage(unsigned char volt_reg, unsigned char value)
 	return 0;
 }
 
+#ifdef CONFIG_PM_PMIC
 static int act8865_set_reg_mode(unsigned char mode_reg, unsigned mode)
 {
 	unsigned char value;
@@ -174,6 +175,7 @@ int act8865_set_power_saving_mode(void)
 
 	return 0;
 }
+#endif
 
 /*--------------------- ACT8865 Workaround -----------------------*/
 
diff --git a/driver/driver_cpp.mk b/driver/driver_cpp.mk
index d04e9b8..54b6022 100644
--- a/driver/driver_cpp.mk
+++ b/driver/driver_cpp.mk
@@ -312,3 +312,7 @@ endif
 ifeq ($(CONFIG_EEPROM_ON_TWI3), y)
 CPPFLAGS += -DCONFIG_EEPROM_ON_TWI3
 endif
+
+ifeq ($(CONFIG_PM_PMIC), y)
+CPPFLAGS += -DCONFIG_PM_PMIC
+endif
diff --git a/driver/pm.c b/driver/pm.c
index 56484d5..c0bde9e 100644
--- a/driver/pm.c
+++ b/driver/pm.c
@@ -30,6 +30,7 @@
 #include "wm8904.h"
 #include "hdmi_SiI9022.h"
 #include "macb.h"
+#include "act8865.h"
 
 void at91_board_pm(void)
 {
@@ -49,4 +50,7 @@ void at91_board_pm(void)
 #ifdef CONFIG_MACB
 	phys_enter_power_down();
 #endif
+#ifdef CONFIG_PM_PMIC
+	act8865_set_power_saving_mode();
+#endif
 }
-- 
1.9.1

